### PR TITLE
Heretic: artifacts can be binded to own keys

### PR DESCRIPTION
**Pull request by JNechaevsky**
By the request of @fabiangreffrath, see: https://github.com/JNechaevsky/russian-doom/commit/d0356943a48c48c7db36a674bd24a908340999c0#commitcomment-36712800

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
This was quick, thanks!

File: src/heretic/g_game.c [hunk 1]
```diff
@@ -453,6 +453,52 @@ void G_BuildTiccmd(ticcmd_t *cmd, int maketic)
...
+    {
+        gamekeydown[key_arti_urn] = false;
+        cmd->arti = arti_superhealth;
+    }
+    else if (gamekeydown[key_arti_bomb] && !cmd->arti)
+    {
+        gamekeydown[key_arti_bomb] = false;
+        cmd->arti = arti_firebomb;
+    }
+    else if (gamekeydown[key_arti_tome] && !cmd->arti)
```
[the remark this answers is not in the document]
Comment (JNechaevsky) [reply src/heretic/g_game.c]: Done, tested (https://github.com/chocolate-doom/chocolate-doom/pull/1236/commits/f7a14366353e4b8f0bb18595234e6833a57f5738). Seems to be correct now?

File: src/heretic/g_game.c [hunk 2]
```diff
@@ -447,12 +447,53 @@ void G_BuildTiccmd(ticcmd_t *cmd, int maketic)
...
     }
-    if (gamekeydown[127] && !cmd->arti
+    if (gamekeydown[key_arti_tome] && !cmd->arti
         && !players[consoleplayer].powers[pw_weaponlevel2])
     {
-        gamekeydown[127] = false;
+        gamekeydown[key_arti_tome] = false;
         cmd->arti = arti_tomeofpower;
     }
+    if (gamekeydown[key_arti_quartz] && !cmd->arti
```
Comment (fabiangreffrath) [inline src/heretic/g_game.c]: Now, if you could add an `else` here. Sorry for being picky.

File: src/m_controls.c [hunk 3]
```diff
@@ -54,6 +54,16 @@ int key_invleft = '[';
 int key_invright = ']';
 int key_useartifact = KEY_ENTER;
 
+int key_arti_quartz = 0;
+int key_arti_urn = 0;
+int key_arti_bomb = 0;
+int key_arti_tome = 0;
```
Comment (fabiangreffrath) [inline src/m_controls.c]: This must be preset to 127 here.

**Comment by mfrancis95** [comment]
What's the need of including `!cmd->arti` in every condition when you can just have that checked once?

**Comment by JNechaevsky** [comment]
@fabiangreffrath, done one, done two: https://github.com/chocolate-doom/chocolate-doom/pull/1236/commits/76fa96e19a7d54993c6df0ffe9076d38655d9ad4

@mfrancis95, no idea, I wasn't seeking for simplifying, it's just replicates an existing code. Worth to change somehow?

**Comment by fabiangreffrath** [comment]
Waif a minute, you pushed the enter key change to the same branch!

**Comment by JNechaevsky** [comment]
Oh st*t! Looks likes I have messed up with commits, sorry for this. Here's a proper commit that I have prepared for PR: https://github.com/JNechaevsky/chocolate-doom/commit/83fed5512b3188bacb60ea6fb38e1e2662860f13

Can we roll back enter key changes somehow?

**Comment by turol** [comment]
You can force-push the older commit to this branch and that will roll it back.

**Comment by JNechaevsky** [comment]
I'm afraid I don't know how to do this. :scream:

**Comment by JNechaevsky** [comment]
Allright, since `master` branch was deleted, now I can't revert "enter key", nothing happens after this: https://github.com/JNechaevsky/chocolate-doom/commit/99e93a9f010f62ff9fff3dd98a06fcfbe468d57c

It's my huge f*ck up, and that's an official reason why I'm trying to avoid to make a PRs to forks with all costs...

**Comment by fabiangreffrath** [comment]
Not *that* bad after all. You could still open a new pull request with the changes in a single commit. But, please, never do pull requests from your repository's master branch, always create a separate branch before.

**Comment by fragglet** [comment]
Heh, whoops. Never mind, the enter key change looks fine to me anyway.

No shame or blame, let's just take it as a learning experience and concentrate on what we can do differently next time. Fabian's comment about always creating a second branch is really good advice and I'm going to make sure I follow it too.

**Comment by fabiangreffrath** [comment]
I have cleaned the mess up a bit. Here are the two new PRs:
https://github.com/chocolate-doom/chocolate-doom/pull/1238
https://github.com/chocolate-doom/chocolate-doom/pull/1237

**Comment by JNechaevsky** [comment]
Thanks for understanding, and please take my apologies for a little bit _too emotional_ lexis. While trying to make a good thing, only mess happens including double amount of work. Honestly, after few years of using Github Desktop I'm pretty fine with working on own repository, but making a PR's between other projects, especially with multiple files, still something unlearned for me.

However, I'm worried about one thing: I'm don't have a Mac at hands and have no idea is there a second "Enter" key, and thus, can't check what will actually happens as a result. My girlfriend have a Macbook Air, but AFAIR there is only one Enter key (a), and she'll slap me on hands for even a simple word about Xcode (b, should never happen!).

**Pull request closed without merging**